Procfile

```diff
@@ -1 +0,0 @@
-web: python -m girder --port ${PORT}
```

Comment:
Given that the `Procfile` is so obviously out of date, I'm going to consider Heroku to now be unsupported via our own neglect, and thus mercilessly remove that section of the docs.

If anyone wants to pick up the baton and fix Heroku deployment (preferably with some sort of testing mechanism), that'd be fine, but until then, we shouldn't (IMHO) document and have cruft for a broken configuration.